feat: provide linux/arm64 platform image

nashtsai opened the pull request:
- upgrade mysql resources to 8.0.32 due to linux/arm64 is only available after 8.0.31

d-bytebase commented:
@cla-bot check

d-bytebase commented:
Super, couldn't wait for this one. Thanks!

Looks like the integration tests are failing.

nashtsai commented on scripts/Dockerfile:
```diff
@@ -5,38 +5,55 @@
 
+ARG NVM_MIRROR=https://mirrors.ustc.edu.cn/node/
+ARG NODE_MIRROR=https://mirrors.ustc.edu.cn/node/
+ARG NPM_REGISTRY=https://npmreg.proxy.ustclug.org/
```
ought to change build script to accept mirrors settings, and shouldn't have default value to CN country settings

nashtsai commented:
> Super, couldn't wait for this one. Thanks!
> 
> Looks like the integration tests are failing.

will look into it

nashtsai commented:
> Super, couldn't wait for this one. Thanks!
> 
> Looks like the integration tests are failing.

I've found that I need to fix missing `backend/resources/postgres/resources_darwin_arm64.go` however, `backend/resources/README.md` didn't provide postgres resources info, could you fix up this README first?

d-bytebase commented:
> > Super, couldn't wait for this one. Thanks!
> > Looks like the integration tests are failing.
> 
> I've found that I need to fix missing `backend/resources/postgres/resources_darwin_arm64.go` however, `backend/resources/README.md` didn't provide postgres resources info, could you fix up this README first?

Give me a day or two. On it.

d-bytebase commented:
A couple comments
1) delete temp files from go generate;
2) split change into multiple PRs.

nashtsai commented:
> A couple comments
> 
> 1. delete temp files from go generate;
> 2. split change into multiple PRs.

I don't see the temp files in this PR, and what is the reason for splitting this PR?

d-bytebase commented:
> > A couple comments
> > 
> > 1. delete temp files from go generate;
> > 2. split change into multiple PRs.
> 
> I don't see the temp files in this PR, and what is the reason for splitting this PR?

Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.

d-bytebase commented on backend/api/v1/auth_service.go:
```diff
@@ -587,7 +587,8 @@ func (s *AuthService) Login(ctx context.Context, request *v1pb.LoginRequest) (*v
 	}
 
-	var accessToken, refreshToken string
+	var accessToken string
```
no need to change this file.

nashtsai replied:
I wouldn't bother the revert this change as it makes not difference.

d-bytebase replied:
It does affect the style. Since it makes no difference, please revert the change to minimize the PR.

d-bytebase commented on .dockerignore:
```diff
@@ -1,3 +1,6 @@
 frontend/node_modules
 backend/server/dist
 bytebase-build
+scripts/Dockerfile*
```
no need to change this file.

nashtsai replied:
take look old version of Dockerfile line 86 `COPY ./scripts /usr/local/bin/` adding this ignore voids unnecessary files, I am keeping it in case others may doing similar act.

d-bytebase replied:
scripts/Dockerfile does matter and we should not ignore.

nashtsai replied:
the syntax is `Dockerfile*` as there are other Dockerfile under this dir., personally I would suggest separate `scirpts` and `Dockerfile(s)`

nashtsai commented:
> > > A couple comments
> > > 
> > > 1. delete temp files from go generate;
> > > 2. split change into multiple PRs.
> > 
> > 
> > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> 
> Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.

I am still not getting the problem here, do elaborate your concerns.

d-bytebase commented:
> > > > A couple comments
> > > > 
> > > > 1. delete temp files from go generate;
> > > > 2. split change into multiple PRs.
> > > 
> > > 
> > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > 
> > 
> > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> 
> I am still not getting the problem here, do elaborate your concerns.

https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide

https://google.github.io/eng-practices/review/developer/small-cls.html

nashtsai commented:
> > > > > A couple comments
> > > > > 
> > > > > 1. delete temp files from go generate;
> > > > > 2. split change into multiple PRs.
> > > > 
> > > > 
> > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > 
> > > 
> > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > 
> > 
> > I am still not getting the problem here, do elaborate your concerns.
> 
> https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> 
> https://google.github.io/eng-practices/review/developer/small-cls.html

After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.

d-bytebase commented:
> > > > > > A couple comments
> > > > > > 
> > > > > > 1. delete temp files from go generate;
> > > > > > 2. split change into multiple PRs.
> > > > > 
> > > > > 
> > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > 
> > > > 
> > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > 
> > > 
> > > I am still not getting the problem here, do elaborate your concerns.
> > 
> > 
> > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > https://google.github.io/eng-practices/review/developer/small-cls.html
> 
> After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.

My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.

nashtsai commented:
> > > > > > > A couple comments
> > > > > > > 
> > > > > > > 1. delete temp files from go generate;
> > > > > > > 2. split change into multiple PRs.
> > > > > > 
> > > > > > 
> > > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > > 
> > > > > 
> > > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > > 
> > > > 
> > > > I am still not getting the problem here, do elaborate your concerns.
> > > 
> > > 
> > > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > > https://google.github.io/eng-practices/review/developer/small-cls.html
> > 
> > 
> > After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.
> 
> My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.

Do advice how you wanna to slit it.

d-bytebase commented:
> > > > > > > > A couple comments
> > > > > > > > 
> > > > > > > > 1. delete temp files from go generate;
> > > > > > > > 2. split change into multiple PRs.
> > > > > > > 
> > > > > > > 
> > > > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > > > 
> > > > > > 
> > > > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > > > 
> > > > > 
> > > > > I am still not getting the problem here, do elaborate your concerns.
> > > > 
> > > > 
> > > > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > > > https://google.github.io/eng-practices/review/developer/small-cls.html
> > > 
> > > 
> > > After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.
> > 
> > 
> > My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.
> 
> Do advice how you wanna to slit it.

One change can include resource filename change plus any additional resource files.
One change for GetTarNameAndVersion().
And the rest.

nashtsai commented:
> > > > > > > > > A couple comments
> > > > > > > > > 
> > > > > > > > > 1. delete temp files from go generate;
> > > > > > > > > 2. split change into multiple PRs.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > > > > 
> > > > > > > 
> > > > > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > > > > 
> > > > > > 
> > > > > > I am still not getting the problem here, do elaborate your concerns.
> > > > > 
> > > > > 
> > > > > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > > > > https://google.github.io/eng-practices/review/developer/small-cls.html
> > > > 
> > > > 
> > > > After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.
> > > 
> > > 
> > > My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.
> > 
> > 
> > Do advice how you wanna to slit it.
> 
> One change can include resource filename change plus any additional resource files. One change for GetTarNameAndVersion(). And the rest.

It's too troublesome for me, if you insist on this, I can merge this PR to a staging branch, and you can arrange your own staff to do it.

d-bytebase commented:
> > > > > > > > > > A couple comments
> > > > > > > > > > 
> > > > > > > > > > 1. delete temp files from go generate;
> > > > > > > > > > 2. split change into multiple PRs.
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > > > > > 
> > > > > > > > 
> > > > > > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > > > > > 
> > > > > > > 
> > > > > > > I am still not getting the problem here, do elaborate your concerns.
> > > > > > 
> > > > > > 
> > > > > > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > > > > > https://google.github.io/eng-practices/review/developer/small-cls.html
> > > > > 
> > > > > 
> > > > > After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.
> > > > 
> > > > 
> > > > My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.
> > > 
> > > 
> > > Do advice how you wanna to slit it.
> > 
> > 
> > One change can include resource filename change plus any additional resource files. One change for GetTarNameAndVersion(). And the rest.
> 
> It's too troublesome for me, if you insist on this, I can merge this PR to a staging branch, and you can arrange your own staff to do it.

The change is less effort than conversation here. Anyway, up to you.

nashtsai commented:
> > > > > > > > > > > A couple comments
> > > > > > > > > > > 
> > > > > > > > > > > 1. delete temp files from go generate;
> > > > > > > > > > > 2. split change into multiple PRs.
> > > > > > > > > > 
> > > > > > > > > > 
> > > > > > > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > I am still not getting the problem here, do elaborate your concerns.
> > > > > > > 
> > > > > > > 
> > > > > > > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > > > > > > https://google.github.io/eng-practices/review/developer/small-cls.html
> > > > > > 
> > > > > > 
> > > > > > After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.
> > > > > 
> > > > > 
> > > > > My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.
> > > > 
> > > > 
> > > > Do advice how you wanna to slit it.
> > > 
> > > 
> > > One change can include resource filename change plus any additional resource files. One change for GetTarNameAndVersion(). And the rest.
> > 
> > 
> > It's too troublesome for me, if you insist on this, I can merge this PR to a staging branch, and you can arrange your own staff to do it.
> 
> The change is less effort than conversation here. Anyway, up to you.

If it's as simple as cherry-pick, I would agree.

d-bytebase commented:
> > > > > > > > > > > > A couple comments
> > > > > > > > > > > > 
> > > > > > > > > > > > 1. delete temp files from go generate;
> > > > > > > > > > > > 2. split change into multiple PRs.
> > > > > > > > > > > 
> > > > > > > > > > > 
> > > > > > > > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > > > > > > > 
> > > > > > > > > > 
> > > > > > > > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > I am still not getting the problem here, do elaborate your concerns.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > > > > > > > https://google.github.io/eng-practices/review/developer/small-cls.html
> > > > > > > 
> > > > > > > 
> > > > > > > After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.
> > > > > > 
> > > > > > 
> > > > > > My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.
> > > > > 
> > > > > 
> > > > > Do advice how you wanna to slit it.
> > > > 
> > > > 
> > > > One change can include resource filename change plus any additional resource files. One change for GetTarNameAndVersion(). And the rest.
> > > 
> > > 
> > > It's too troublesome for me, if you insist on this, I can merge this PR to a staging branch, and you can arrange your own staff to do it.
> > 
> > 
> > The change is less effort than conversation here. Anyway, up to you.
> 
> If it's as simple as cherry-pick, I would agree.

Copy the files out to a directory (from your branch), copy it back, and make a new PR. Should just take a few minutes.

I will be okay if you have the resource change and GetTarNameAndVersion change in a single PR.

nashtsai commented:
> > > > > > > > > > > A couple comments
> > > > > > > > > > > 
> > > > > > > > > > > 1. delete temp files from go generate;
> > > > > > > > > > > 2. split change into multiple PRs.
> > > > > > > > > > 
> > > > > > > > > > 
> > > > > > > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > I am still not getting the problem here, do elaborate your concerns.
> > > > > > > 
> > > > > > > 
> > > > > > > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > > > > > > https://google.github.io/eng-practices/review/developer/small-cls.html
> > > > > > 
> > > > > > 
> > > > > > After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.
> > > > > 
> > > > > 
> > > > > My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.
> > > > 
> > > > 
> > > > Do advice how you wanna to slit it.
> > > 
> > > 
> > > One change can include resource filename change plus any additional resource files. One change for GetTarNameAndVersion(). And the rest.
> > 
> > 
> > It's too troublesome for me, if you insist on this, I can merge this PR to a staging branch, and you can arrange your own staff to do it.
> 
> The change is less effort than conversation here. Anyway, up to you.

As I've stated, it's too troublesome for me.

d-bytebase commented:
> > > > > > > > > > > > A couple comments
> > > > > > > > > > > > 
> > > > > > > > > > > > 1. delete temp files from go generate;
> > > > > > > > > > > > 2. split change into multiple PRs.
> > > > > > > > > > > 
> > > > > > > > > > > 
> > > > > > > > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > > > > > > > 
> > > > > > > > > > 
> > > > > > > > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > I am still not getting the problem here, do elaborate your concerns.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > > > > > > > https://google.github.io/eng-practices/review/developer/small-cls.html
> > > > > > > 
> > > > > > > 
> > > > > > > After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.
> > > > > > 
> > > > > > 
> > > > > > My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.
> > > > > 
> > > > > 
> > > > > Do advice how you wanna to slit it.
> > > > 
> > > > 
> > > > One change can include resource filename change plus any additional resource files. One change for GetTarNameAndVersion(). And the rest.
> > > 
> > > 
> > > It's too troublesome for me, if you insist on this, I can merge this PR to a staging branch, and you can arrange your own staff to do it.
> > 
> > 
> > The change is less effort than conversation here. Anyway, up to you.
> 
> As I've stated, it's too troublesome for me.

I suggest to split the PR because I'm pretty sure the current change in Dockerfile will break our underlying infrastructure such as demo.bytebase.com, etc. Then we don't need to revert the whole PR as the rest of the change looks fine.

The other way is to move the rest of change (everything not in the resources) into a separate PR.

We will not merge the current PR if it doesn't pass code review.

nashtsai commented:
> > > > > > > > > > > > > A couple comments
> > > > > > > > > > > > > 
> > > > > > > > > > > > > 1. delete temp files from go generate;
> > > > > > > > > > > > > 2. split change into multiple PRs.
> > > > > > > > > > > > 
> > > > > > > > > > > > 
> > > > > > > > > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > > > > > > > > 
> > > > > > > > > > > 
> > > > > > > > > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > > > > > > > > 
> > > > > > > > > > 
> > > > > > > > > > I am still not getting the problem here, do elaborate your concerns.
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > > > > > > > > https://google.github.io/eng-practices/review/developer/small-cls.html
> > > > > > > > 
> > > > > > > > 
> > > > > > > > After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.
> > > > > > > 
> > > > > > > 
> > > > > > > My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.
> > > > > > 
> > > > > > 
> > > > > > Do advice how you wanna to slit it.
> > > > > 
> > > > > 
> > > > > One change can include resource filename change plus any additional resource files. One change for GetTarNameAndVersion(). And the rest.
> > > > 
> > > > 
> > > > It's too troublesome for me, if you insist on this, I can merge this PR to a staging branch, and you can arrange your own staff to do it.
> > > 
> > > 
> > > The change is less effort than conversation here. Anyway, up to you.
> > 
> > 
> > As I've stated, it's too troublesome for me.
> 
> I suggest to split the PR because I'm pretty sure the current change in Dockerfile will break our underlying infrastructure such as demo.bytebase.com, etc. Then we don't need to revert the whole PR as the rest of the change looks fine.
> 
> The other way is to move the rest of change (everything not in the resources) into a separate PR.
> 
> We will not merge the current PR if it doesn't pass code review.

Then do comment required changes for Dockerfile.

d-bytebase commented:
> > > > > > > > > > > > > > A couple comments
> > > > > > > > > > > > > > 
> > > > > > > > > > > > > > 1. delete temp files from go generate;
> > > > > > > > > > > > > > 2. split change into multiple PRs.
> > > > > > > > > > > > > 
> > > > > > > > > > > > > 
> > > > > > > > > > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > > > > > > > > > 
> > > > > > > > > > > > 
> > > > > > > > > > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > > > > > > > > > 
> > > > > > > > > > > 
> > > > > > > > > > > I am still not getting the problem here, do elaborate your concerns.
> > > > > > > > > > 
> > > > > > > > > > 
> > > > > > > > > > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > > > > > > > > > https://google.github.io/eng-practices/review/developer/small-cls.html
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.
> > > > > > > 
> > > > > > > 
> > > > > > > Do advice how you wanna to slit it.
> > > > > > 
> > > > > > 
> > > > > > One change can include resource filename change plus any additional resource files. One change for GetTarNameAndVersion(). And the rest.
> > > > > 
> > > > > 
> > > > > It's too troublesome for me, if you insist on this, I can merge this PR to a staging branch, and you can arrange your own staff to do it.
> > > > 
> > > > 
> > > > The change is less effort than conversation here. Anyway, up to you.
> > > 
> > > 
> > > As I've stated, it's too troublesome for me.
> > 
> > 
> > I suggest to split the PR because I'm pretty sure the current change in Dockerfile will break our underlying infrastructure such as demo.bytebase.com, etc. Then we don't need to revert the whole PR as the rest of the change looks fine.
> > The other way is to move the rest of change (everything not in the resources) into a separate PR.
> > We will not merge the current PR if it doesn't pass code review.
> 
> Then do comment required changes for Dockerfile.

Sorry, I don't do further review unless we split the PR. Thanks.

nashtsai commented:
> > > > > > > > > > > > > > > A couple comments
> > > > > > > > > > > > > > > 
> > > > > > > > > > > > > > > 1. delete temp files from go generate;
> > > > > > > > > > > > > > > 2. split change into multiple PRs.
> > > > > > > > > > > > > > 
> > > > > > > > > > > > > > 
> > > > > > > > > > > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > > > > > > > > > > 
> > > > > > > > > > > > > 
> > > > > > > > > > > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > > > > > > > > > > 
> > > > > > > > > > > > 
> > > > > > > > > > > > I am still not getting the problem here, do elaborate your concerns.
> > > > > > > > > > > 
> > > > > > > > > > > 
> > > > > > > > > > > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > > > > > > > > > > https://google.github.io/eng-practices/review/developer/small-cls.html
> > > > > > > > > > 
> > > > > > > > > > 
> > > > > > > > > > After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > Do advice how you wanna to slit it.
> > > > > > > 
> > > > > > > 
> > > > > > > One change can include resource filename change plus any additional resource files. One change for GetTarNameAndVersion(). And the rest.
> > > > > > 
> > > > > > 
> > > > > > It's too troublesome for me, if you insist on this, I can merge this PR to a staging branch, and you can arrange your own staff to do it.
> > > > > 
> > > > > 
> > > > > The change is less effort than conversation here. Anyway, up to you.
> > > > 
> > > > 
> > > > As I've stated, it's too troublesome for me.
> > > 
> > > 
> > > I suggest to split the PR because I'm pretty sure the current change in Dockerfile will break our underlying infrastructure such as demo.bytebase.com, etc. Then we don't need to revert the whole PR as the rest of the change looks fine.
> > > The other way is to move the rest of change (everything not in the resources) into a separate PR.
> > > We will not merge the current PR if it doesn't pass code review.
> > 
> > 
> > Then do comment required changes for Dockerfile.
> 
> Sorry, I don't do further review unless we split the PR. Thanks.

@d-bytebase Just let you know, that I've just left ApeCloud and I made this contribution as there were plans to provider ByteBase as a KubeBlocks' addon, anyway this work will make no progress, I shall close this PR accordingly.

d-bytebase commented:
> > > > > > > > > > > > > > > > A couple comments
> > > > > > > > > > > > > > > > 
> > > > > > > > > > > > > > > > 1. delete temp files from go generate;
> > > > > > > > > > > > > > > > 2. split change into multiple PRs.
> > > > > > > > > > > > > > > 
> > > > > > > > > > > > > > > 
> > > > > > > > > > > > > > > I don't see the temp files in this PR, and what is the reason for splitting this PR?
> > > > > > > > > > > > > > 
> > > > > > > > > > > > > > 
> > > > > > > > > > > > > > Just for the risk of being rollback. We don't need to rollback the whole change to 42 files.
> > > > > > > > > > > > > 
> > > > > > > > > > > > > 
> > > > > > > > > > > > > I am still not getting the problem here, do elaborate your concerns.
> > > > > > > > > > > > 
> > > > > > > > > > > > 
> > > > > > > > > > > > https://github.com/bytebase/bytebase/blob/main/docs/code-review-guide.md#code-review-guide
> > > > > > > > > > > > https://google.github.io/eng-practices/review/developer/small-cls.html
> > > > > > > > > > > 
> > > > > > > > > > > 
> > > > > > > > > > > After going through small CLs, this PR doesn't violate the rules, and plz go through this PR carefully, I am make necessary adjustments as a whole.
> > > > > > > > > > 
> > > > > > > > > > 
> > > > > > > > > > My suggestion is moving this PR piece by piece. A lot of the changes under resources/ LGTM. If we could split the PR, that's already checked-in.
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > Do advice how you wanna to slit it.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > One change can include resource filename change plus any additional resource files. One change for GetTarNameAndVersion(). And the rest.
> > > > > > > 
> > > > > > > 
> > > > > > > It's too troublesome for me, if you insist on this, I can merge this PR to a staging branch, and you can arrange your own staff to do it.
> > > > > > 
> > > > > > 
> > > > > > The change is less effort than conversation here. Anyway, up to you.
> > > > > 
> > > > > 
> > > > > As I've stated, it's too troublesome for me.
> > > > 
> > > > 
> > > > I suggest to split the PR because I'm pretty sure the current change in Dockerfile will break our underlying infrastructure such as demo.bytebase.com, etc. Then we don't need to revert the whole PR as the rest of the change looks fine.
> > > > The other way is to move the rest of change (everything not in the resources) into a separate PR.
> > > > We will not merge the current PR if it doesn't pass code review.
> > > 
> > > 
> > > Then do comment required changes for Dockerfile.
> > 
> > 
> > Sorry, I don't do further review unless we split the PR. Thanks.
> 
> @d-bytebase Just let you know, that I've just left ApeCloud and I made this contribution as there were plans to provider ByteBase as a KubeBlocks' addon, anyway this work will make no progress, I shall close this PR accordingly.

Thanks for the contribution and best attempt. There's a lot more work to do for providing an ARM based image, and we will take over and continue the project once we have more resources. Appreciate all the work here.
